pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -122,18 +120,28 @@ in buildPythonPackage rec {
     repo   = "pytorch";
     rev    = "v${version}";
     fetchSubmodules = true;
-    sha256 = "1biyq2p48chakf2xw7hazzqmr5ps1nx475ql8vkmxjg5zaa071cz";
+    sha256 = "07ga9806c5q02hsdybwdw9nc0r1a8dinwjakyg6wlrccclc027lq";
   };
 
-  dontUseCmakeConfigure = true;
-
   preConfigure = lib.optionalString cudaSupport ''
     export TORCH_CUDA_ARCH_LIST="${lib.strings.concatStringsSep ";" final_cudaArchList}"
     export CC=${cudatoolkit.cc}/bin/gcc CXX=${cudatoolkit.cc}/bin/g++
   '' + lib.optionalString (cudaSupport && cudnn != null) ''
     export CUDNN_INCLUDE_DIR=${cudnn}/include
   '';
 
+  # Use pytorch's custom configurations
+  dontUseCmakeConfigure = true;
+
+  BUILD_NAMEDTENSOR = true;
+  BUILD_DOCS = buildDocs;
+  USE_MKLDNN = mklSupport;
+  preBuild = with lib.strings; concatStringsSep "\n" [
+    ''export CMAKE_PREFIX_PATH="$CMAKE_PREFIX_PATH:${if mklSupport then mkl else openblas}"''
+    "python setup.py build --cmake-only"
```

Comment:
```${python.interpreter}```

Are wheels not yet supported?

Reply:
Wheels are supported! That's how pytorch is distributed with pip and various configurations can be found at [pytorch.org](https://pytorch.org).  I just always assumed that building from source was the preferred nix way.

Is there a good example which depends on third-party C/C++-libraries you can point me to? searching the github repository seems noisy.

The python section makes it look like I should pass `format="wheel"` -- is it possible to simply do this and fetch the source with `fetchPypi`?

Reply:
In general yes, it's preferred to build from source. On reason to keep the source build is to be able to configure the package for things like CUDA or changing blas implementation. Also, tests are usually distributed with the source distribution.

But if you can correctly link a wheel, and it works as intended then that's definitely a possible route.

Reply:
I meant *building* a wheel.

Reply:
but I see now it's just a preBuild step

Reply:
oh, you mean `python setup.py build bdist_wheel`, xD